librosa/feature/spectral.py

```diff
@@ -2007,4 +2047,4 @@ def melspectrogram(
     # Build a Mel filter
     mel_basis = filters.mel(sr, n_fft, **kwargs)
 
-    return np.dot(mel_basis, S)
+    return np.einsum("...ft,mf->...mt", S, mel_basis,optimize=True)
```

Comment:
mfcc LGTM for now; we may want to revisit the idea of exposing power_to_db params at the API here.